object-detections-api.yaml

```diff
@@ -0,0 +1,78 @@
+swagger: "2.0"
+info:
+  description: "Extremely Simple And Portable Objects Detection API"
+  version: "0.0.1"
+  title: "Embedded Objects Detector And Classifier"
+  termsOfService: "https://cardiffdevlunch.io/terms/"
+  contact:
+    email: "product@cardiffdevlunch.io"
+  license:
+    name: "Beerware (rev. 42)"
+    url: "https://spdx.org/licenses/Beerware.html"
+host: "sentinel.cardiffdevlunch.io"
+basePath: "/v1"
+tags:
+- name: "Detection"
+  description: "Objects detection"
+schemes:
+- "http"
+paths:
+  /detectObjects:
+    post:
+      tags:
+        - Detection API
+      summary: Performs object detection on uploaded image
+      description: 'Performs simple and low accuracy object detection to help filter-out irrelevant parts of the video stream'
+      operationId: uploadFile
+      consumes:
+        - multipart/form-data
+      produces:
+        - application/json
+      parameters:
+        - name: image
+          in: formData
+          description: image to upload in order to perform object detection
+          required: true
+          type: file
+        - name: imageResolution
+          description: image resolution (Width x Height) in pixels
+          in: formData
+          required: true
+          type: string
+          enum: ["800x600", "640x480"]
+        - name: imageFormat
+          description: allowed image formats
+          in: formData
+          type: string
+          required: true
+          enum: ["jpeg", "png"]
+      responses:
+        '200':
+          description: successful operation
+          schema:
+            $ref: '#/definitions/ApiResponse'
+        '405':
+          description: "Invalid input parameters"
+        '415':
+          description: "Unsupported media type. Only JPEG/PNG image formats are supported"
+        '503': 
+          description: "Service Unavailable - Neural Network's meltdown"
+        '504':
+          description: "Takes too much time to process an image. Back-off and try again later"

```

Comment:
what should be considered too much time? will back off help?

Reply:
That was proposal for you to have an ability to throttle consumer's requests. If :noidea: what do to with that on your side - let's just drop it.

Reply:
lets just apply natural back-pressure for now ;)